Add jei tooltip and fix bug 添加jei tooltip和修复暮色森林联动的生物转化配方

burin-ljx opened the pull request:
修改如下：
之前tooltip只会显示方块名称，现在会显示方块名称、方块id、配方id和模组名称

<img width="1920" height="1051" alt="2025-09-19_20 54 49" src="https://github.com/user-attachments/assets/2dce3c3c-41cd-481c-8c48-98e7bb4564b0" />

<img width="1920" height="1051" alt="2025-09-19_20 54 53" src="https://github.com/user-attachments/assets/6daedd69-3a6a-4797-9a3c-0a0cd5553f31" />

- Fixed #2765

burin-ljx committed:
feat(recipe): 添加暮色森林MOD支持

- 新增暮色森林MOD中的僵尸转换配方
- 修复了暮色森林MOD不加载导致的错误
- 优化了配方的显示和工具提示

YocyCraft commented on src/main/java/dev/dubhe/anvilcraft/util/TooltipUtil.java:
```diff
@@ -0,0 +1,36 @@
+package dev.dubhe.anvilcraft.util;
+
+import mezz.jei.api.helpers.IModIdHelper;
+import mezz.jei.common.Internal;
```
`mezz.jei.common.Internal`类的代码文档：
> For JEI internal use only, these are normally accessed from the API.

虽然该类被标记为`public`，但此类并非对外开放的API，这意味着代码随时可能随版本更新修改，存在版本不兼容的风险。

burin-ljx committed:
refactor(util): 重构 TooltipUtil以移除对 JEI 的直接依赖

移除了对 mezz.jei 相关类的导入和使用，改为通过 NeoForge 的 ModList 和 ModContainer获取模组名称。新增 getModName 方法用于根据 modId 获取显示名称或格式化后的名称。此更改解耦了 TooltipUtil与 JEI 的直接关联，提高代码通用性和可维护性。

Gu-ZT committed:
Merge remote-tracking branch 'upstream/releases/1.21' into pr/2766

Gu-ZT committed:
feat(integration): 提取暮色森林模组兼容代码至独立包

将原本分散在各 RecipeLoader 中的暮色森林模组兼容逻辑统一迁移至
`dev.dubhe.anvilcraft.integration.twilight` 包下，创建独立的集成类
`TwilightIntegration` 进行管理。同时删除原 RecipeLoader 类中相关
的导入和条件编译注解，简化原有代码结构，提升可维护性与模块清晰度。

此外，清理了部分冗余的 import 语句，并优化了部分方法实现方式，如使用switch 表达式替代连续 if 判断等。
